src/util/fetchUser.ts

```diff
@@ -0,0 +1,13 @@
+import fetch from "node-fetch";
+import BSUser from "../constants/bsUser";
+import urls from "../constants/urls";
+
+async function fetchUser(username: string): Promise<null | BSUser> {
+  const url = urls.BINARYSEARCH_API.replace("USERNAME", username);
+  const data = await fetch(url).then(res => res.json());
+
+  if ("user" in data) return data;
```

Comment:
the `"profile"` field in the payload is not that important, should just `return data.user`

Reply:
got it